src/core/reactor.cc

```diff
@@ -1073,6 +1075,166 @@ void reactor::start_handling_signal() {
 
 namespace internal {
```

Comment:
Vs how I put it in `reactor.cc`? No reason per-say, I was copying how they went about implementing the stall detector. Normally I would've put the whole implementation in `cpu_profiler.cc` so I'm definitely willing to move things over.

Reply:
Ok, I would move it but at the end of the day something seastar people can decide on.

Reply:
Yeah, I think I will just move it for now when I go to split the main commit up into smaller commits.

Reply:
Yeah I agree it's good to move it over, also reactor.cc is just a massive file the compiled version of which already makes tools like `addr2line` since it has so much debug info etc.

Reply:
The code has now been moved to a separate file.